### PR TITLE
Initial PSRule without parameter files #206

### PR DESCRIPTION
**Pull request by BernieWhite**
# Overview/Summary

Adds integration with PSRule for Azure for testing repository code with Azure Well-Architected Framework.

Working towards #206 

## This PR fixes/adds/changes/removes

1. Added PSRule integration using VSCode and GitHub Actions. Parameter files/ Bicep tests not included.

### Breaking Changes

N/A

## Testing Evidence

Modules resources are not being processed yet until we have snippets to expand them.

```text
Rules processed: 124, failed: 0, errored: 0
Run Azure/ALZ-Bicep/2170788848 completed in 00:00:07.1681704
```

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [ ] Associated it with relevant [ADO items](https://dev.azure.com/unifiedactiontracker/Solution%20Engineering/_backlogs/backlog/Azure%20Landing%20Zone%20Bicep/Backlog%20Bugs%20Feedback)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.

### CODE REVIEW

**Comment by BernieWhite** [comment]
@jtracey93 Initial effort without parameter files. Which I understood that you didn't want merged yet, because we would work towards snippets for deployment/ testing. Let me know if there is anything else you feel we need to do to get this PR merged.

File: ps-rule.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  # Exclude Bicep docs files
+  - docs/scripts/callModuleFromACR.example.bicep
+
+  # Exclude Bicep module files
+  - 'infra-as-code/bicep/modules/**/*.bicep'
+  - 'infra-as-code/bicep/CRML/**/*.bicep'
+
+  # Include test files from modules
+  - '!infra-as-code/bicep/modules/**/*.tests.bicep'
```
Comment (jtracey93) [inline ps-rule.yaml:51]: Is this in the correct part of the config file?

Or is this saying "don't ignore" because of the `!`?

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: ps-rule.yaml (hunk 1 above)
Comment (BernieWhite) [reply ps-rule.yaml:51]: @jtracey93 Correct. It follows the git path spec. Entries are normally ignored. Entries beginning with `!` are re-included if previously ignored.

This is the standard example, but I can remove it until we work through where the snippets will be located.

Comment (jtracey93) [reply ps-rule.yaml:51]: Awesome thanks @BernieWhite.

Can we tweak it to look for `.bicep` files in every module folder called `samples` and then inside this folder we will create the new "samples/tests" that PSRule can scan.

We can then get this merged into the `ps-rule` branch and then start adding the samples in and making the adjustments required and also solves another item for including samples bicep files as well 👍

**Review by jtracey93 — APPROVED** [review]
LGTM 🚢

**Pull request merged**